prow/upgrade-tests.sh

```diff
@@ -25,42 +27,39 @@ WD=$(dirname "$0")
 WD=$(cd "$WD"; pwd)
 ROOT=$(dirname "$WD")
 
-# Local run example:
-# UPGRADE_TEST_LOCAL="true" HUB=gcr.io/istio-release \
-#   ISTIO_REL_URL=https://storage.googleapis.com/istio-release-pipeline-data/daily-build/master-20190107-09-15 \
-#   TAG=master-20190107-09-15 \
-#   prow/upgrade-tests.sh
-
-# Set up inputs needed by test_upgrade.sh
-export SOURCE_HUB=${SOURCE_HUB:-istio}
-export SOURCE_VERSION=1.2.5
-export SOURCE_RELEASE_PATH="https://github.com/istio/istio/releases/download/${SOURCE_VERSION}"
-export TARGET_HUB=${TARGET_HUB:-$HUB}
-export TARGET_VERSION=${TAG}
-export TARGET_RELEASE_PATH=${ISTIO_REL_URL}
+# Set up inputs needed by /istio/istio/tests/upgrade/test_crossgrade.sh, the default settings is to test upgrade from
+# 1.3.1 to 1.3.2 using helm chart
+# These environment variables are passed by /istio/test-infra/prow/cluster/jobs istio periodic upgrade jobs
+export SOURCE_HUB=${SOURCE_HUB:-"docker.io/istio"}
+export SOURCE_TAG=${SOURCE_TAG:-"1.3.1"}
+export SOURCE_RELEASE_PATH=${SOURCE_RELEASE_PATH:-"https://github.com/istio/istio/releases/download/${SOURCE_TAG}"}
+export TARGET_HUB=${TARGET_HUB:-"docker.io/istio"}
+export TARGET_TAG=${TAGET_TAG:-"1.3.2"}
+export TARGET_RELEASE_PATH=${TAGET_RELEASE_PATH:-"https://github.com/istio/istio/releases/download/${TARGET_TAG}"}
+export INSTALL_OPTIONS=${ISTALL_OPTIONS:-"helm"}
+export FROM_PATH=${FROM_PATH:-"$(mktemp -d from_dir.XXXXXX)"}
+export TO_PATH=${TO_PATH:-"$(mktemp -d to_dir.XXXXXX)"}
+
+# Update Environment Variables for upgrade from 1.3 to 1.4-latest
+if "${TEST_TARGET}"=1.4; then
+  BRANCH="${TEST_TARGET}-dev"
+  TARGET_TAG=$(curl "https://storage.googleapis.com/istio-build/dev/${BRANCH}")
+  TARGET_RELEASE_PATH="https://gcsweb.istio.io/gcs/istio-build/dev/${TARGET_TAG}"
+fi
 
 # Set to any non-empty value to use kubectl configured cluster instead of mason provisioned cluster.
-UPGRADE_TEST_LOCAL="${UPGRADE_TEST_LOCAL:-}"
-
-# This is a script to download release artifacts from monthly or daily release
-# location and kick off upgrade/downgrade tests.
-#
-# This is currently triggered by https://github.com/istio-releases/daily-release
-# for release qualification.
-#
-# Expects HUB, SOURCE_VERSION, TARGET_VERSION, SOURCE_RELEASE_PATH, and TARGET_RELEASE_PATH as inputs.
+UPGRADE_TEST_LOCAL="${UPGRADE_TEST_LOCAL:-"true"}"
```

Comment:
I think you probably either need this or kind. We want kind at some point but right now looks like we have nothing

Reply:
yes, later on we will use kind